xl2times/transforms.py

```diff
@@ -50,48 +50,60 @@
 
 def remove_comment_rows(
     config: datatypes.Config,
-    table: datatypes.EmbeddedXlTable,
+    tables: List[datatypes.EmbeddedXlTable],
     model: datatypes.TimesModel,
-) -> datatypes.EmbeddedXlTable:
+) -> List[datatypes.EmbeddedXlTable]:
     """
-    Return a modified copy of 'table' where rows with cells starting with symbols
-    indicating a comment row in any column have been deleted. Comment row symbols
-    are column name dependant and are specified in the config.
-
-    :param table:       Table object in EmbeddedXlTable format.
-    :return:            Table object in EmbeddedXlTable format without comment rows.
+    Remove comment rows from all the tables. Assumes table dataframes are not empty.
     """
-    if table.dataframe.size == 0:
-        return table
+    result = []
 
-    df = table.dataframe.copy()
+    for table in tables:
+        tag = datatypes.Tag(table.tag.split(":")[0])
+        if tag in config.row_comment_chars:
+            df = table.dataframe
+            remove_df_comment_rows(df, config.row_comment_chars[tag])
+            # Keep the table if it stays not empty
+            if not df.empty:
+                result.append(table)
+            else:
+                continue
+        else:
+            result.append(table)
 
-    tag = table.tag.split(":")[0]
+    return result
 
-    if tag in config.row_comment_chars:
-        chars_by_colname = config.row_comment_chars[tag]
-    else:
-        return table
+
+def remove_df_comment_rows(
```

Comment:
```suggestion
def _remove_df_comment_rows(
```
I think the convention is to start a function name with `_` if it is a helper function that isn't used outside of this file